ios/RCTWebRTC/WebRTCModule+Transceivers.m

```diff
@@ -178,6 +185,72 @@ @implementation WebRTCModule (Transceivers)
     resolve(@true);
 }
 
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(transceiverSetCodecPreferences
+                                       : (nonnull NSNumber *)objectID senderId
+                                       : (NSString *)senderId codecPreferences
+                                       : (NSArray *)codecPreferences) {
+    RTCPeerConnection *peerConnection = self.peerConnections[objectID];
+
+    if (peerConnection == nil) {
+        RCTLogWarn(@"PeerConnection %@ not found in transceiverSetCodecPreferences()", objectID);
+        return nil;
+    }
+
+    RTCRtpTransceiver *transceiver = nil;
+    for (RTCRtpTransceiver *t in peerConnection.transceivers) {
+        if ([senderId isEqual:t.sender.senderId]) {
+            transceiver = t;
+            break;
+        }
+    }
+
+    if (transceiver == nil) {
+        RCTLogWarn(@"senderSetParameters() transceiver is null");
+        return nil;
+    }
+
+    // Convert JSON codec capabilities to the actual objects.
+    RTCRtpTransceiverDirection direction = transceiver.direction;
+    NSMutableArray *availableCodecs = [NSMutableArray new];
+    if (direction == RTCRtpTransceiverDirectionSendRecv || direction == RTCRtpTransceiverDirectionSendOnly) {
+        RTCRtpCapabilities *capabilities = [self.peerConnectionFactory rtpSenderCapabilitiesFor:transceiver.mediaType];
+        for (RTCRtpCodecCapability *codec in capabilities.codecs) {
+            NSDictionary *codecDict = [SerializeUtils codecCapabilityToJSON:codec];
+            [availableCodecs addObject:@{
+                @"dict" : codecDict,
+                @"codec" : codec,
+            }];
+        }
+    }
+    if (direction == RTCRtpTransceiverDirectionSendRecv || direction == RTCRtpTransceiverDirectionRecvOnly) {
+        RTCRtpCapabilities *capabilities =
+            [self.peerConnectionFactory rtpReceiverCapabilitiesFor:transceiver.mediaType];
+        for (RTCRtpCodecCapability *codec in capabilities.codecs) {
+            NSDictionary *codecDict = [SerializeUtils codecCapabilityToJSON:codec];
+            [availableCodecs addObject:@{
+                @"dict" : codecDict,
+                @"codec" : codec,
+            }];
+        }
+    }
+
+    // Codec preferences is order sensitive.
+    NSMutableArray *codecsToSet = [NSMutableArray new];
+
+    for (NSDictionary *codecDict in codecPreferences) {
+        for (NSDictionary *entry in availableCodecs) {
+            NSDictionary *availableCodecDict = [entry objectForKey:@"dict"];
+            if ([codecDict isEqualToDictionary:availableCodecDict]) {
+                [codecsToSet addObject:[entry objectForKey:@"codec"]];
+                break;
+            }
+        }
+    }
+
+    transceiver.codecPreferences = codecsToSet;
```

Comment:
@davidliu Looks like we need to use setCodecPreferences here, but it takes RTCRtpCapabilities so some of this conversion is not necessary I think?

Reply:
Yeah, I think I might have changed the code compared to when the initial draft pr was made, so I'll need to recheck this commit. The M118 build is available to try out for ios?

Reply:
Yep feel free to add commits to this PR!

Reply:
`setCodecPreferences` takes RTCRtpCodecCapabilities rather than just RTCRtpCapabilities, so the conversion is needed. Fixed up the usage, so it compiles now and should be working as intended.